### ISSUE TITLE
Xcode 12.2 cannot compile for iOS Simulator (No CMAKE_CXX_COMPILER could be found)

### ISSUE THREAD

**Issue by chrisballinger** [opencv/opencv#18789]
##### System information (version)

- OpenCV => 4.5.0 (master)
- Operating System / Platform => Catalina
- Compiler => Xcode 12.2

##### Detailed description

Xcode 12.2 cannot be used to compile the iOS Simulator architectures (either `arm64` for Apple Silicon or `x86_64` for Intel macs). I believe this is an issue in CMake itself (not fixed as of `3.19.0-rc3`): https://gitlab.kitware.com/cmake/cmake/-/issues/21425

edit: will be resolved in cmake 3.19.0, workaround available here https://gitlab.kitware.com/cmake/cmake/-/merge_requests/5506

```
Executing: cmake -GXcode -DAPPLE_FRAMEWORK=ON -DCMAKE_INSTALL_PREFIX=install -DCMAKE_BUILD_TYPE=Release -DOPENCV_INCLUDE_INSTALL_PATH=include -DOPENCV_3P_LIB_INSTALL_PATH=lib/3rdparty -DFRAMEWORK_NAME=opencv2 -DBUILD_opencv_objc=OFF -DIOS_ARCH=arm64 -DCMAKE_TOOLCHAIN_FILE=/Users/chrisbal/Documents/opencv/platforms/ios/cmake/Toolchains/Toolchain-iPhoneSimulator_Xcode.cmake -DCMAKE_SYSTEM_PROCESSOR=arm64 -DCMAKE_OSX_ARCHITECTURES=arm64 -DCPU_BASELINE=DETECT -DCMAKE_CROSSCOMPILING=ON -DOPENCV_WORKAROUND_CMAKE_20989=ON /Users/chrisbal/Documents/opencv
-- Setting up iPhoneSimulator toolchain for IOS_ARCH='arm64'
-- iPhoneSimulator toolchain loaded
-- Setting up iPhoneSimulator toolchain for IOS_ARCH='arm64'
-- iPhoneSimulator toolchain loaded
-- The CXX compiler identification is AppleClang 12.0.0.12000032
-- The C compiler identification is AppleClang 12.0.0.12000032
CMake Error at CMakeLists.txt:109 (enable_language):
  No CMAKE_CXX_COMPILER could be found.



CMake Error at CMakeLists.txt:109 (enable_language):
  No CMAKE_C_COMPILER could be found.



-- Configuring incomplete, errors occurred!
See also "/Users/chrisbal/Documents/opencv/platforms/apple/xcframework-build/iphonesimulator-arm64/build/build-arm64-iphonesimulator/CMakeFiles/CMakeOutput.log".
```

##### Steps to reproduce

1. Download and install Xcode 12.2 RC: https://download.developer.apple.com/Developer_Tools/Xcode_12.2_Release_Candidate/Xcode_12.2_Release_Candidate.xip

```
$ sudo xcode-select -s /Applications/Xcode-beta.app/
$ cd path/to/opencv
$ python2 ./platforms/ios/build_framework.py --iphoneos_archs arm64 --iphonesimulator_archs x86_64 .
```

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues,
       answers.opencv.org, Stack Overflow, etc and have not found solution
   
 - [x] I updated to latest OpenCV version and the issue is still there
   
 - [x] There is reproducer code and related data files: videos, images, onnx, etc

**Comment by alalek** [comment]
Could you please attach related error messages from `CMakeFiles/CMakeOutput.log` / `CMakeFiles/CMakeError.log`?
(it would help other users to find this issue in case of similar problems)

**Comment by chrisballinger** [comment]
Workaround now available if you want to patch your existing cmake install until 3.19.0 is out with the fix: https://gitlab.kitware.com/cmake/cmake/-/merge_requests/5506

Here's the [CMakeOutput.log](https://github.com/opencv/opencv/files/5532492/CMakeOutput.log), there actually isn't an "error" because it succeeds according to the log, it just fails a regex check in the cmake internals because of a change in Xcode 12.2.

**Comment by jenjia** [comment]
> Workaround now available if you want to patch your existing cmake install until 3.19.0 is out with the fix: https://gitlab.kitware.com/cmake/cmake/-/merge_requests/5506

Where you able to compile opencv2 4.5 with cmake 3.19.0 ?

In my case, I downloaded 3.19.0-rc3 , the cmake error you described went away, but now all the ..Ext.swift files fail to compile.

os: 11.0.1 (Big Sur) , XCode: 12.2 , OpenCV: 4.5

**Comment by kikaxa** [comment]
> Were you able to compile opencv2 4.5 with cmake 3.19.0 ?
I was able to compile 4.5 with a patched cmake 3.18.

**Comment by jenjia** [comment]
> > Were you able to compile opencv2 4.5 with cmake 3.19.0 ?
> 
> I was able to compile 4.5 with a patched cmake 3.18.

Well, my errors persist I will need to make a new ticket about it.

Anyways, this can be closed CMake released 3.19.0 which fixes this problem.

**Comment by chrisballinger** [comment]
CMake 3.19 is out with the fix: https://cmake.org/cmake/help/latest/release/3.19.html
